src/librustc_resolve/lib.rs

```diff
@@ -5115,7 +5115,7 @@ impl<'a> Resolver<'a> {
                         // extra for the comma.
                         span.lo().0 - (prev_comma.as_bytes().len() as u32) - 1
                     ));
-                    err.span_suggestion(
+                    err.tool_only_span_suggestion(
```

Comment:
I'm unsure about this change. I agree that it makes the error much cleaner, but without the `help: ...` message how would someone know that they can automatically have the problem fixed?

Reply:
`tool_only_span_diagnostics` was landed over the all hands because it was needed for a different change, so it's pretty recent.

Tools still will see the same json output, so vscode and rustfix would still be able to apply these. I guess I hadn't thought of the use case of someone running rustfix after running rustc iteratively. Should we mark these some other way? Cyan asterisk next to the error or something?

Reply:
My only concern is that users who know when they see `help: ...` that it is something that can be automatically applied won't know that for this error. 

Something like a cyan asterisk would be ideal to draw attention to that (it might even raise awareness of rustfix being able to fix these things if everyone's error messages had a cyan asterisk suddenly appear). I wouldn't block this PR on that, so I'll r+ this and we can file a follow-up.

Reply:
Filed #59103.